config/v2/validation.go

```diff
@@ -41,229 +41,80 @@ func (c *Config) Validate() error {
 	errs = multierror.Append(errs, c.validateInheritedStringField("backend region", BackendRegionGetter, nonEmptyString))
 	errs = multierror.Append(errs, c.validateInheritedStringField("backend profile", BackendProfileGetter, nonEmptyString))
 
-	errs = multierror.Append(errs, c.validateInheritedStringField("provider region", AWSProviderRegionGetter, nonEmptyString))
-	errs = multierror.Append(errs, c.validateInheritedStringField("provider profile", AWSProviderProfileGetter, nonEmptyString))
-	errs = multierror.Append(errs, c.validateInheritedStringField("provider version", AWSProviderVersionGetter, nonEmptyString))
-	errs = multierror.Append(errs, c.validateInheritedIntField("provider account id", AWSProviderAccountIdGetter, nonZeroInt64))
-
+	errs = multierror.Append(errs, c.ValidateAWSProviders())
 	errs = multierror.Append(errs, c.validateModules())
 
 	return errs.ErrorOrNil()
 }
 
-func nonEmptyString(s string) bool {
-	return len(s) > 0
-}
-
-func OwnerGetter(comm Common) *string {
-	return comm.Owner
-}
-
-func ProjectGetter(comm Common) *string {
-	return comm.Project
-}
-
-func TerraformVersionGetter(comm Common) *string {
-	return comm.TerraformVersion
-}
-
-func BackendBucketGetter(comm Common) *string {
-	if comm.Backend != nil {
-		return comm.Backend.Bucket
-	}
-	return nil
-}
-func BackendRegionGetter(comm Common) *string {
-	if comm.Backend != nil {
-		return comm.Backend.Region
-	}
-	return nil
-}
-
-func BackendDynamoTableGetter(comm Common) *string {
-	if comm.Backend != nil {
-		return comm.Backend.DynamoTable
-	}
-	return nil
-}
-
-func BackendProfileGetter(comm Common) *string {
-	if comm.Backend != nil {
-		return comm.Backend.Profile
-	}
-	return nil
-}
-
-func AWSProviderRegionGetter(comm Common) *string {
-	if comm.Providers != nil && comm.Providers.AWS != nil {
-		return comm.Providers.AWS.Region
-
-	}
-	return nil
-}
-
-func AWSProviderVersionGetter(comm Common) *string {
-	if comm.Providers != nil && comm.Providers.AWS != nil {
-		return comm.Providers.AWS.Version
-	}
-	return nil
-}
-
-func AWSProviderProfileGetter(comm Common) *string {
-	if comm.Providers != nil && comm.Providers.AWS != nil {
-		return comm.Providers.AWS.Profile
-	}
-	return nil
-}
-
-func AWSProviderAccountIdGetter(comm Common) *int64 {
-	if comm.Providers != nil && comm.Providers.AWS != nil {
-		return comm.Providers.AWS.AccountID
-	}
-	return nil
-}
-
-func AWSProviderAdditionalRegionsGetter(comm Common) []string {
-	if comm.Providers != nil && comm.Providers.AWS != nil {
-		return comm.Providers.AWS.AdditionalRegions
-	}
-	return nil
-}
-
-func ExtraVarsGetter(comm Common) map[string]string {
-	if comm.ExtraVars != nil {
-		return comm.ExtraVars
-	}
-	return map[string]string{}
-}
+func (c *Config) ValidateAWSProviders() error {
+	var errs *multierror.Error
 
-func nonZeroInt64(i int64) bool {
-	return true
-}
+	var validate = func(p AWSProvider, component string) error {
```

Comment:
I wonder if should be `AWSProvider.Validate() error`

Reply:
I am refactoring this a bit i the next PR (for snowflake provider). It is useful to have it as a function to keep the code generic.

Otherwise I think to take this approach I would have to create an `Validatable` interface or some such.